profiles/common/modules/features/nexteuropa_newsroom/nexteuropa_newsroom.blocks.inc

```diff
@@ -0,0 +1,889 @@
+  module_load_include('inc', 'nexteuropa_newsroom', 'classes/NewsroomSummaryBlock');
+  module_load_include('inc', 'nexteuropa_newsroom', 'classes/NewsroomBlock');
+  module_load_include('inc', 'nexteuropa_newsroom', 'classes/NewsroomBlockAgenda');
+  module_load_include('inc', 'nexteuropa_newsroom', 'classes/NewsroomAgendaBlockSchedule');
+
+  $summary_block = new NewsroomSummaryBlock();
+  return $summary_block->generateContent();
+}
+
+/**
+ * Gets related content tree.
+ *
+ * @return string
+ *   Content.
+ */
+function _nexteuropa_nexteuropa_related_content() {
+  $node = menu_get_object();
+  $content = '';
+  if ($node) {
+    $children_related_items = _nexteuropa_nexteuropa_get_related_content('field_related_content', $node);
+    $parent_related_items = _nexteuropa_nexteuropa_get_related_content('field_newsroom_parent_content', $node);
+    $parent_nid = count($parent_related_items) > 0 ? key($parent_related_items) : NULL;
+    if ($parent_related_items || $children_related_items) {
+      $content = theme('newsroom_related_content', array(
+        'parent_items' => $parent_related_items,
+        'current_item' => $node,
+        'brother_items' => _nexteuropa_newsroom_get_related_children('field_newsroom_parent_content', $parent_nid),
+        'children_items' => $children_related_items,
+      ));
+    }
+  }
+  return $content;
+}
+
+/**
+ * Try to resolve each taxonomy term with a corresponding ID to map.
+ *
+ * @param string $field_name
+ *   The mapping field.
+ * @param string $nid
+ *   The node id.
+ *
+ * @return array
+ *   Taxonomy ids.
+ */
+function _nexteuropa_newsroom_get_related_children($field_name, $nid) {
+  // Let's initiate term_ids to field value, in case we can't map it.
+  $items = array();
+  if ($nid) {
+    $query = new EntityFieldQuery();
+    $query->entityCondition('entity_type', 'node')
+        ->fieldCondition($field_name, 'target_id', $nid)
+        ->execute();
+    $result = $query->execute();
+    if (isset($result['node'])) {
+      $ids = array_keys($result['node']);
+
+      foreach ($ids as $id) {
+        $items[$id] = _nexteuropa_newsroom_get_node_title($id);
+      }
+    }
+  }
+  return $items;
+}
+
+/**
+ * Get related content from node.
+ *
+ * @param string $field_name
+ *   Field name.
+ * @param object $node
+ *   Current node.
+ *
+ * @return array
+ *   Related nodes.
+ */
+function _nexteuropa_nexteuropa_get_related_content($field_name, $node) {
+  $items = array();
+  $field = isset($node->$field_name) ? $node->$field_name : NULL;
+  if ($field && isset($field[LANGUAGE_NONE])) {
+    foreach ($field[LANGUAGE_NONE] as $item) {
+      $items[$item['target_id']] = isset($item['entity']) ? $item['entity']->title : _nexteuropa_newsroom_get_node_title($item['target_id']);
+    }
+  }
+  return $items;
+}
+
+/**
+ * Helper function to get the node title without loading the whole node object.
+ */
+function _nexteuropa_newsroom_get_node_title($nid) {
```

Comment:
It would be great to have entity everywhere, but unfortunately it is not a case. So to avoid loading of a full entity we load only titles. That is why we have additional method and cache here is justified.